mdtk/degradations.py

```diff
@@ -36,6 +38,56 @@ def seeded_func(*args, seed=seed, **kwargs):
     return seeded_func
 
 
+def pre_process(df, sort=False):
+    """
+    Function which will pre-process a dataframe to be degraded.
+    
+    Currently, that means resetting the indices to consecutive ints from 0.
+    Optionally, this will sort the df (depending on the degradation).
+    This function is called automatically by each degradation.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        The dataframe to pre-process
+
+    sort : boolean
+        True to sort the dataframe. Flase to leave the ordering as given.
+
+    Returns
+    -------
+    df : pd.DataFrame
+        The postprocessed dataframe.
+    """
+    if sort:
+        df = df.sort_values(NOTE_DF_SORT_ORDER)
+    df = df.reset_index(drop=True)
+    return df
```

Comment:
I'm cool with this, with the view that, in `make_dataset.py`, we shouldn't need to use it: preproc should all happen when making the `Composition` object.

Reply:
That's true. We could potentially add a pre_process flag to each degradation, defaulting to True.